Kafka/src/main/java/io/deephaven/kafka/publish/GenericRecordKeyOrValueSerializer.java

```diff
@@ -0,0 +1,723 @@
+            if (previous) {
+                doubleContext.inputChunk = chunkSource.getPrevChunk(doubleContext.getContext, keys).asDoubleChunk();
+            } else {
+                doubleContext.inputChunk = chunkSource.getChunk(doubleContext.getContext, keys).asDoubleChunk();
+            }
+
+            for (int ii = 0; ii < doubleContext.inputChunk.size(); ++ii) {
+                final double raw = doubleContext.inputChunk.get(ii);
+                if (raw == QueryConstants.NULL_DOUBLE) {
+                    avroChunk.get(ii).put(fieldName, null);
+                } else {
+                    avroChunk.get(ii).put(fieldName, raw);
+                }
+            }
+        }
+    }
+
+    private class ObjectFieldProcessor<T> extends GenericRecordFieldProcessor {
+        private final ColumnSource chunkSource;
+
+        public ObjectFieldProcessor(String fieldName, ColumnSource chunkSource) {
+            super(fieldName);
+            this.chunkSource = chunkSource;
+        }
+
+        private class ObjectContext implements FieldContext {
+            ChunkSource.GetContext getContext;
+            ObjectChunk inputChunk;
+
+            ObjectContext(int size) {
+                getContext = chunkSource.makeGetContext(size);
+            }
+
+            @Override
+            public void close() {
+                getContext.close();
+            }
+        }
+
+        @Override
+        FieldContext makeContext(int size) {
+            return new ObjectContext(size);
+        }
+
+        @Override
+        void processField(FieldContext fieldContext, WritableObjectChunk<GenericRecord, Attributes.Values> avroChunk,
+                OrderedKeys keys, boolean previous) {
+            final ObjectContext objectContext = (ObjectContext) fieldContext;
+            if (previous) {
+                objectContext.inputChunk = chunkSource.getPrevChunk(objectContext.getContext, keys).asObjectChunk();
+            } else {
+                objectContext.inputChunk = chunkSource.getChunk(objectContext.getContext, keys).asObjectChunk();
+            }
+
+            for (int ii = 0; ii < objectContext.inputChunk.size(); ++ii) {
+                final Object raw = objectContext.inputChunk.get(ii);
+                avroChunk.get(ii).put(fieldName, raw);
+            }
+        }
+    }
+
+    private class TimestampFieldProcessor extends GenericRecordFieldProcessor {
+        public TimestampFieldProcessor(String fieldName) {
+            super(fieldName);
+        }
+
+        @Override
+        FieldContext makeContext(int size) {
+            return null;
+        }
+
+        @Override
+        public void processField(FieldContext fieldContext,
+                WritableObjectChunk<GenericRecord, Attributes.Values> avroChunk, OrderedKeys keys, boolean isRemoval) {
+            // do we really want nanos instead of micros; there are avro things for micros/millis?
```

Comment:
There are, to do this right seems like we would need to consider the field type in the schema?
https://avro.apache.org/docs/1.8.2/spec.html#Timestamp+%28millisecond+precision%29

Reply:
Also, we need to handle DBDateTime outside of the auto generated at publish time (ie, as an actual existing column of type DBDateTime) as one of those avro types too?